lib/relayer/RelayTransactionSigner.js

```diff
@@ -0,0 +1,118 @@
+const GasPriceOracle = require('./GasPriceOracle')
+
+const DEFAULT_DOMAIN_NAME = 'Aragon Relayer'
+const DEFAULT_DOMAIN_VERSION = '1'
+
+const GAS_MULTIPLIER = 1.18
+
+const FIRST_TX_GAS_OVERLOAD = 83500
+const NORMAL_TX_GAS_OVERLOAD = 53500
+
+const DATA_TYPES = {
+  EIP712Domain: [
+    { name: 'name', type: 'string' },
+    { name: 'version', type: 'string' },
+    { name: 'chainId', type: 'uint256' },
+    { name: 'verifyingContract', type: 'address' },
+  ],
+  Transaction: [
+    { name: 'to', type: 'address' },
+    { name: 'nonce', type: 'uint256' },
+    { name: 'data', type: 'bytes' },
+    { name: 'gasRefund', type: 'uint256' },
+    { name: 'gasPrice', type: 'uint256' }
+  ],
+}
+
+module.exports = web3 => class RelayTransactionSigner {
```

Comment:
What do you think about either placing the web3 instance inside these classes, or extracting the web3 instance used into its own util (e.g. [getWeb3()](https://github.com/aragon/aragon/blob/master/src/web3-utils.js#L134)).

Generally we don't run services in the context of a truffle environment (maybe we should? I'm a bit scared at needing to depend on them for the service to stay alive), so there's no global web3 instance available.

-------

It'd also avoid defining a class in a function, which is a bit awkward, since you can have different definitions if you invoke the function more than once. This then starts to violate the `instanceof` checks and prototype chains, and gets very awkward if you ever need to compare / manipulate more than one object derived from what should be same class.

Reply:
Completely agree, it should be part of the constructor o at least a known object that resolves that dependency